### ISSUE TITLE
Istio service mesh blocks outbound http traffic because of wrong service name

### ISSUE THREAD

**Issue by oba11** [cowboysysop/charts#33]
## Issue

Deployed istio 1.6 version and noticed that external https requests were failing from my pod

<details>
  <summary>Click to expand!</summary>

```
curl -Iv https://api.example.com/_health
* Expire in 0 ms for 6 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 0 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 4 ms for 1 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 1 ms for 1 (transfer 0x56328e019f50)
* Expire in 4 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 2 ms for 1 (transfer 0x56328e019f50)
* Expire in 8 ms for 1 (transfer 0x56328e019f50)
* Expire in 4 ms for 1 (transfer 0x56328e019f50)
* Expire in 4 ms for 1 (transfer 0x56328e019f50)
* Expire in 8 ms for 1 (transfer 0x56328e019f50)
* Expire in 5 ms for 1 (transfer 0x56328e019f50)
* Expire in 5 ms for 1 (transfer 0x56328e019f50)
* Expire in 8 ms for 1 (transfer 0x56328e019f50)
* Expire in 7 ms for 1 (transfer 0x56328e019f50)
* Expire in 7 ms for 1 (transfer 0x56328e019f50)
* Expire in 8 ms for 1 (transfer 0x56328e019f50)
* Expire in 9 ms for 1 (transfer 0x56328e019f50)
* Expire in 9 ms for 1 (transfer 0x56328e019f50)
* Expire in 12 ms for 1 (transfer 0x56328e019f50)
*   Trying 1.2.3.4...
* TCP_NODELAY set
* Expire in 149980 ms for 3 (transfer 0x56328e019f50)
* Expire in 200 ms for 4 (transfer 0x56328e019f50)
* Connected to api.example.com (1.2.3.4) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: none
  CApath: /etc/ssl/certs
* TLSv1.3 (OUT), TLS handshake, Client hello (1):
* error:1408F10B:SSL routines:ssl3_get_record:wrong version number
* Closing connection 0
curl: (35) error:1408F10B:SSL routines:ssl3_get_record:wrong version number
```
</details>

Rolled back the istio upgrade and I noticed this error in the `istiod` proxy status

```
2020-06-10T05:45:06.283908Z info  ads Push Status: {
    "pilot_conflict_outbound_listener_http_over_https": {
        "vpa-webhook.kube-system.svc.cluster.local": {
            "proxy": "pod-f857f6ffb-hhzx6.staging",
            "message": "listener conflict detected: service vpa-webhook.kube-system.svc.cluster.local specifies an HTTP service on HTTPS only port 443."
        }
    },
    "pilot_conflict_outbound_listener_tcp_over_current_tcp": {},
    "pilot_eds_no_instances": {}
}
```

## Resolution
Manually updating the admission-controller service port name to `https` https://github.com/cowboysysop/charts/blob/5c6b0a1b3041935da58a13fa2becd1e79bee6fb0/charts/vertical-pod-autoscaler/templates/admission-controller/service.yaml#L15

**Comment by sebastien-prudhomme** [comment]
Hi @oba11 , can you explain me why Istio considers that a simple Service name ``http``, that could be ``ẁeb``, ``api`` or ``ẁhatever``, means the protocol is HTTP?

If i refer to this PR istio/istio#17515, the protocol HTTP is defined in a ServiceEntry resource, not a Service resource.

Does Istio automatically create an Istio ServiceEntry from each Service resource in the cluster?

**Comment by sebastien-prudhomme** [comment]
@oba11 I've looked how other chart developers resolve this problem, particularly in the Bitnami chart repository.

I will do another PR to rename all ports in all my charts to make them compatible with Istio.

When it will be in beta phase (perhaps K8S 1.19) , i think I will add ```appProtocol``` on all services.

**Comment by sebastien-prudhomme** [comment]
Fixed in #36
